Result/ResultType.swift

```diff
@@ -101,3 +101,29 @@ infix operator &&& {
 public func &&& <L: ResultType, R: ResultType where L.Error == R.Error> (left: L, @autoclosure right: () -> R) -> Result<(L.Value, R.Value), L.Error> {
 	return left.flatMap { left in right().map { right in (left, right) } }
 }
+
+/// Return a array of `V` by applying transform to all elements of array 'U' and filtering out failure ones
+public func mapFilter<U, V, Error>(results: [U], @noescape transform: U -> Result<V, Error>) -> [V]{
+	var vs = [V]()
+	for result in results{
+		if case let .Success(value) = transform(result){
+			vs.append(value)
+		}
+	}
+	return vs
+}
+
+/// Return a Result with an array of `V`s if all Results of applying `transform` are `Success`es or return the error of the first `Failure`
+public func mapM<U, V, Error>(results: [U], @noescape transform: U -> Result<V, Error>) -> Result<[V], Error>{
```

Comment:
A little further on this, and similar to [the discussion of `mapFilter` above](https://github.com/antitypical/Result/pull/133/files#r48482363), we can probably decouple the essential action here, that of producing a success with all values if they are all successful, or failing otherwise, from the incidental action of mapping using `transform`, to get a slightly more idiomatic take on it:

```swift
public extension CollectionType where Generator.Element: ResultType {
    public func all() -> Result<[Generator.Element.Value], Generator.Element.Error> {
        …
    }
}
```

`mapM`’s full behaviour can be recovered by chaining after `CollectionType.map`:

```swift
let results = values.map(validate).all()
```

And you can use `lazy`, as well.